packages/react/src/utils/useIntersectionRef.ts

```diff
@@ -30,12 +30,9 @@ export default function useIntersectionRef<T extends HTMLElement>(
         return;
       }
 
-      if (
```

Comment:
I think the missing piece from this function is we might not be resolving the ref correctly prior to passing it into the use effect. We might need to instead resolve the element prior to passing it into the dependency array, that way we're always dealing with either an `HTMLElement` or `null`.

Reply:
Added element resolving